src/templates/add-payment-method-block/index.vue

```diff
@@ -0,0 +1,268 @@
+          color: '#ededed'
+        },
+        invalid: {
+          color: '#fa755a',
+          iconColor: '#fa755a'
+        }
+      }
+    }
+    cardNumber.value = stripeComponents.value.create('cardNumber', { ...inputStyles, showIcon: true })
+    cardExpiry.value = stripeComponents.value.create('cardExpiry', inputStyles)
+    cardCvc.value = stripeComponents.value.create('cardCvc', inputStyles)
+  }
+
+  const addStripeComponentsToTemplate = () => {
+    cardNumber.value.mount('#card-number-element')
+    cardExpiry.value.mount('#card-expiry-element')
+    cardCvc.value.mount('#card-cvc-element')
+  }
+
+  const visibleDrawer = computed({
+    get: () => props.visible,
+    set: (value) => {
+      changeVisisbleDrawer(value, true)
+    }
+  })
+
+  const changeVisisbleDrawer = (isVisible) => {
+    emit('update:visible', isVisible)
+  }
+
+  const toggleDrawerVisibility = (isVisible) => {
+    visibleDrawer.value = isVisible
+  }
+
+  const closeDrawer = () => {
+    toggleDrawerVisibility(false)
+  }
+
+  const showToast = (severity, summary) => {
+    const options = {
+      closable: true,
+      severity: severity,
+      summary: severity,
+      detail: summary
+    }
+
+    toast.add(options)
+  }
+
+  const handleSubmit = async () => {
+    isSubmitting.value = true
+
+    const { token, error: submitionErrors } = await stripe.value.createToken(cardNumber.value, {
+      name: cardholderName.value
+    })
+    if (submitionErrors !== undefined) {
+      showToast('error', submitionErrors.message)
+    }
+    if (submitionErrors === undefined) {
+      const accountData = accountStore.account
+      const payload = {
+        card_address_zip: accountData.postal_code,
+        card_country: accountData.country,
+        stripe_token: token.id,
+        card_id: token.card.id,
+        card_brand: token.card.brand,
+        card_holder: token.card.name,
+        card_last_4_digits: token.card.last4,
+        card_expiration_month: token.card.exp_month,
+        card_expiration_year: token.card.exp_year
+      }
+      try {
+        const response = await props.createService(payload)
+        emit('onSuccess', response)
+        showToast('success', response.feedback)
+        showGoBack.value = props.showBarGoBack
+        toggleDrawerVisibility(false)
+      } catch (error) {
+        emit('onError', error)
+        showToast('error', error)
+      }
+    }
+    isSubmitting.value = false
+  }
+
+  const handleGoBack = () => {
+    showGoBack.value = false
+    toggleDrawerVisibility(false)
+  }
+</script>
+
+<template>
+  <Sidebar
+    v-model:visible="visibleDrawer"
+    :update:visible="toggleDrawerVisibility"
+    position="right"
+    :pt="{
+      root: { class: 'max-w-4xl w-full' },
+      headercontent: { class: 'flex justify-content-between items-center w-full pr-2' },
+      content: { class: 'p-8' }
+    }"
+  >
+    <template #header>
+      <h2>Add Payment Method</h2>
+      <FeedbackFish />
```

Comment:
Does it make sense to add FeedbackFish? We are actually thinking about getting rid of it